module/Motion/Rotation.cpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file   Rotation.cpp
+ * @brief  回頭動作の中間クラス
+ * @author desty505
+ */
+
+#include "Rotation.h"
+
+using namespace std;
+
+Rotation::Rotation(double _targetSpeed, bool _isClockwise)
+  : targetSpeed(_targetSpeed), isClockwise(_isClockwise)
+{
+}
+
+void Rotation::run()
+{
+  // 事前条件を判定する
+  if(!isMetPrecondition(targetSpeed)) {
+    return;
+  }
+
+  // isClockwiseがtrueなら時計回り，falseなら反時計回り
+  int leftSign = isClockwise ? 1 : -1;
+  int rightSign = isClockwise ? -1 : 1;
+
+  // 呼び出し時の走行距離
+  double leftMileage = Mileage::calculateWheelMileage(Measurer::getLeftCount());
+  double rightMileage = Mileage::calculateWheelMileage(Measurer::getRightCount());
+
+  // 両輪が目標距離に到達するまでループ
+  while(true) {
+    // 事後条件を判定する
+    if(!isMetPostcondition(leftMileage, rightMileage)) break;
```

Comment:
①は対応してほしいです。②は好みです。

①事後条件は「満たしたから終了！」って条件ではなくて、「終了時に満たしてるから正常終了」って条件だと思うから、終了条件(もしくは継続条件)みたいに使うのは違うと思う。

②好みの話になるけど、whileの継続条件に入れてよくない？(postconditionを満たさない間継続)
場所の話で関数名は①で指摘したとおり事後条件ではないと思う。
```suggestion
  while(!isMetPostcondition(leftMileage, rightMileage)) {
```

Reply:
質問を返してしまって申し訳ないのですが、

ここで必要なのは、
事後条件と終了条件のどちらなのか分かってないのですが、
どちらがいいと思いますか？

Reply:
終了条件(もしくは継続条件)がいいと思う。
事前条件定義してるから事後条件ある方がそれっぽいけど、検証すべき事後条件あんまり思いつかない。